mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreRightManager.java

```diff
@@ -168,11 +172,14 @@ public void setRights(MailboxPath mailboxPath, MailboxACL mailboxACL, MailboxSes
         MailboxMapper mapper = mailboxSessionMapperFactory.getMailboxMapper(session);
         Mailbox mailbox = mapper.findMailboxByPath(mailboxPath);
 
-        setRights(mailboxACL, mapper, mailbox);
+        setRights(mailboxACL, mapper, mailbox, session);
     }
 
-    private void setRights(MailboxACL mailboxACL, MailboxMapper mapper, Mailbox mailbox) throws MailboxException {
+    private void setRights(MailboxACL mailboxACL, MailboxMapper mapper, Mailbox mailbox, MailboxSession session) throws MailboxException {
         mapper.execute(Mapper.toTransaction(() -> mapper.setACL(mailbox, mailboxACL)));
+
+        ACLDiff aclDiff = ACLDiff.computeDiff(mailbox.getACL(), mailboxACL);
+        dispatcher.aclLookupAdded(session, mailbox.generateAssociatedPath(), mailbox.getMailboxId(), aclDiff);
     }
 
```

Comment:
You still don't fire it when **updateRights**?

Isn't it required to use the above tests?

Reply:
I fire them all: in applyRightsCommand and two functions which updateRights (setRights by mailboxPath and mailboxId)